plugins/module_utils/utils.py

```diff
@@ -26,3 +28,174 @@ def generate_api_endpoint(path, **kwargs):
+
+    Sample Existing Data:
+    ---------------------
+    existing_data = {
+        "bfdMultiHopPol": {"adminState": "enabled", "minRxInterval": 250},
+        "bfdPol": {"adminState": "enabled", "detectionMultiplier": 3},
+        "name": "irp_1",
+    }
+    ops = []
+    update_path = "/tenantPolicyTemplate/template/l3OutIntfPolGroups/0"
+
+    replace_data = {
+        ("name"): "new_name",
+        "description": "new_description",
+        ("bfdMultiHopPol", "ifControl"): dict(),
+        ("bfdMultiHopPol", "ifControl", "adminState"): "enabled",
+        ("ospfIntfPol"): dict(ifControl=dict(adminState="disbaled"), cost=0),
+    }
+
+    append_update_ops_data(ops, existing_data, interface_routing_policy_path, replace_data)
+
+    Replace function call output data:
+    ----------------------------------
+
+    Standard Output Data:
+    ---------------------
+    {
+        "bfdMultiHopPol": {
+            "adminState": "enabled",
+            "minRxInterval": 250,
+            "ifControl": {"adminState": "enabled"},
+        },
+        "bfdPol": {
+            "adminState": "enabled",
+            "detectionMultiplier": 3,
+        },
+        "name": "new_name",
+        "description": "new_description",
+        "ospfIntfPol": {
+            "ifControl": {
+                "adminState": "disbaled",
+            },
+            "cost": 0,
+        },
+    }
+
+    API Input Data:
+    ---------------
+    [
+        {"op": "replace", "path": "/tenantPolicyTemplate/template/l3OutIntfPolGroups/1/name", "value": "new_name"},
+        {"op": "replace", "path": "/tenantPolicyTemplate/template/l3OutIntfPolGroups/1/description", "value": "new_description"},
+        {"op": "replace", "path": "/tenantPolicyTemplate/template/l3OutIntfPolGroups/1/bfdMultiHopPol/ifControl", "value": {}},
+        {"op": "replace", "path": "/tenantPolicyTemplate/template/l3OutIntfPolGroups/1/bfdMultiHopPol/ifControl/adminState", "value": "enabled"},
+        {
+            "op": "replace",
+            "path": "/tenantPolicyTemplate/template/l3OutIntfPolGroups/1/ospfIntfPol",
+            "value": {"ifControl": {"adminState": "disbaled"}, "cost": 0},
+        },
+    ]
+
+    ----------------------------------------------------------------
+
+    Sample data for the remove call:
+    --------------------------------
+
+    existing_data = {
+        "bfdMultiHopPol": {
+            "adminState": "enabled",
+            "minRxInterval": 250,
+            "ifControl": {"adminState": "enabled"},
+        },
+        "bfdPol": {
+            "adminState": "enabled",
+            "detectionMultiplier": 3,
+        },
+        "name": "new_name",
+        "description": "new_description",
+        "ospfIntfPol": {
+            "ifControl": {
+                "adminState": "disbaled",
+            },
+            "cost": 0,
+        },
+    }
+    ops = []
+    update_path = "/tenantPolicyTemplate/template/l3OutIntfPolGroups/0"
+
+    remove_data = {
```

Comment:
Do we need this to be a dictionary, because we only require the keys for removal?

```suggestion
    remove_data = [key1, key2]
```